test/serverExplorer.test.ts

```diff
@@ -0,0 +1,193 @@
+        expect(refreshStub).calledOnce;
+        expect(getChildren).deep.equals(undefined);
+    });
+
+    test('removeServer call should remove server from tree data model', () => {
+        const disposeStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            clear: () => {},
+            dispose: () => {}
+        });
+        const getChildren = serverExplorer.getChildren(null);
+        sandbox.stub(serverExplorer, 'refresh');
+        serverExplorer.removeServer(serverHandle);
+        expect(disposeStub).calledOnce;
+        expect(getChildren).deep.equals(undefined);
+    });
+
+    test('serverExplorer.updateServer call should update server state to received in state change event', () => {
+        const serviceStub = sandbox.stub(serverExplorer.servers, 'get').returns({
+            stateChange
+        });
+        const clearStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            clear: () => {
+                return true;
+            }
+        });
+        sandbox.stub(serverExplorer, 'refresh');
+        const updateserver = serverExplorer.updateServer(stateChange);
+        expect(serviceStub).calledOnce;
+        expect(clearStub).calledOnce;
+        expect(clearStub).calledOnceWith(updateserver);
+    });
+
+    test('serverExplorer.showOutput call should show servers output channel', () => {
+        const showStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            show: () => {}
+        });
+        serverExplorer.showOutput(serverHandle);
+        expect(showStub).calledOnce;
+
+    });
+
+    test('serverExplorer.getTreeItem call should return TreeItem instance with corresponding label', () => {
+        const serverHandleStub = sandbox.stub(serverExplorer.serverStatus, 'get').resolves({
+            serverHandle
+        });
+        serverExplorer.getTreeItem(serverHandle);
+        expect(serverHandleStub).calledOnce;
+    });
+
+    test('should able to getChildren of the server', () => {
+        const clearStub = sandbox.stub(serverExplorer.servers, 'values').returns([]);
+        const getChildren = serverExplorer.getChildren(undefined);
+        expect(getChildren).deep.equals(Array.from(clearStub));
+    });
+
+    test('serverExplorer.addServerOutput call should show ServerOutput', () => {
+        const getStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns(undefined);
+        serverExplorer.addServerOutput(ProcessOutput);
+        expect(getStub).calledOnce;
+    });
+
+    test('serverExplorer.addServerOutput call should show ServerOutput', () => {
+        const getStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            show: () => {},
+            append: () => {}
+        });
+        serverExplorer.addServerOutput(ProcessOutput);
+        expect(getStub).calledOnce;
+    });
+
+    test('serverExplorer.refresh call should able check server refresh', () => {
```

Comment:
Name should be: `serverExplorer.refresh() should trigger serverExporer.getChildren() call for root node`. It means getChildren() is called with undefined passed as value for parent node and that exactly what test should verify.